### PR TITLE
Southxchange support

### PR DESCRIPTION
**Pull request by filenko45**

### CODE REVIEW

File: freqtrade/data/converter.py [hunk 1]
```diff
@@ -73,6 +73,9 @@ def ohlcv_fill_up_missing_data(dataframe: DataFrame, timeframe: str, pair: str)
         'volume': 'sum'
     }
     ticker_minutes = timeframe_to_minutes(timeframe)
+
+    if exchange == 'southxchange':
+        return dataframe
```
Comment (Krestol) [inline freqtrade/data/converter.py]: I'm not shure it is a right behavior.

File: freqtrade/exchange/common.py [hunk 2]
```diff
@@ -69,7 +69,7 @@
         'negociecoins',
         'nova',
         'paymium',
-        'southxchange',
+        #'southxchange',
```
Comment (Krestol) [inline freqtrade/exchange/common.py]: remove instead of comment

File: freqtrade/exchange/southxchange.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+from freqtrade.exchange import Exchange
+
+logger = logging.getLogger(__name__)
+
+
+class Southxchange(Exchange):
+
+    def __init__(self, config: dict, validate: bool = True) -> None:
+        super().__init__(config, validate)
+        for i in self.markets:
```
Comment (Krestol) [inline freqtrade/exchange/southxchange.py]: Just add default values like here https://github.com/Krestol/ccxt/blob/graviex_python_support/python/ccxt/graviex.py#L162

File: freqtrade/exchange/southxchange.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        :return: None
+        """
+
+        if not self.markets:
+            logger.warning("Unable to validate pairs (assuming they are correct).")
+            return
+
+        for pair in pairs:
+            # Note: ccxt has BaseCurrency/QuoteCurrency format for pairs
+            # TODO: add a support for having coins in BTC/USDT format
```
Comment (Krestol) [inline freqtrade/exchange/southxchange.py]: is it necessary comments?

File: freqtrade/freqtradebot.py [hunk 5]
```diff
@@ -373,6 +374,33 @@ def execute_buy(self, pair: str, stake_amount: float, price: Optional[float] = N
         order_id = order['id']
         order_status = order.get('status', None)
 
+        if order_status is None:
+            order['status'] = 'open'
+            order['cost'] = stake_amount
```
Comment (Krestol) [inline freqtrade/freqtradebot.py:381]: why cost == stake_amount?
amount * price ?

File: freqtrade/freqtradebot.py [hunk 6]
```diff
@@ -488,7 +516,7 @@ def get_real_amount(self, trade: Trade, order: Dict, order_amount: float = None)
         if order_amount is None:
             order_amount = order['amount']
         # Only run for closed orders
-        if trade.fee_open == 0 or order['status'] == 'open':
+        if trade.fee_open == 0 or order.get('status', None) == 'open':
```
Comment (Krestol) [inline freqtrade/freqtradebot.py]: is it possible for an order to be with undefined status?

File: freqtrade/data/converter.py (hunk 1 above)
Comment (filenko45) [reply freqtrade/data/converter.py]: we don't need to fill missing data for southxchange

File: freqtrade/freqtradebot.py (hunk 5 above)
Comment (DancingAxolotl) [reply freqtrade/freqtradebot.py:381]: stake_amount is used as parameter to api so the same value is saved in the order. Also, the amount is calculated above as amount = stake_amount / buy_limit_requested so it's simply reverse.

File: freqtrade/freqtradebot.py (hunk 6 above)
Comment (DancingAxolotl) [reply freqtrade/freqtradebot.py]: Impossible now, this must be leftover from some earlier changes

File: freqtrade/freqtradebot.py [hunk 7]
```diff
@@ -919,8 +968,42 @@ def execute_sell(self, trade: Trade, limit: float, sell_reason: SellType) -> Non
...
+            order['cost'] = limit * trade.amount
+            order['price'] = limit
+            order['amount'] = trade.amount
+            order['side'] = 'sell'
+            order['type'] = 'limit'
+            order['is_open'] = True
+            order['filled'] = 0
+
+        if order.get('fee', None) is None:
+            order['fee'] = {'cost': fee, 'currency': trade.pair[:trade.pair.find('/')]}
```
Comment (Krestol) [inline freqtrade/freqtradebot.py]: self.exchange.get_fee()

Comment (DancingAxolotl) [reply freqtrade/freqtradebot.py]: exchange.get_fee returns only 'rate' for some reason

File: freqtrade/freqtradebot.py (hunk 5 above)
Comment (Krestol) [reply freqtrade/freqtradebot.py:381]: \+

File: freqtrade/freqtradebot.py (hunk 7 above)
Comment (Krestol) [reply freqtrade/freqtradebot.py]: ok, so we can use exchange._api.calculate_fee

**Pull request merged**